src/shared/osclib/__init__.py

```diff
@@ -152,115 +152,44 @@ def are_on_same_machine(url1: str, url2: str) -> bool:
     except BaseException:
         return False
 
-    hostname_1, hostname_2 = address1.hostname, address2.hostname
-    if hostname_1 == hostname_2:
+    if address1.hostname == address2.hostname:
         return True
 
-    # check if address is IPv6 (between brackets)
-    addr1_ipv6, addr2_ipv6 = False, False
-    if hostname_1.startswith('[') and hostname_1.endswith(']'):
-        hostname_1 = hostname_1[1:-1]
-        addr1_ipv6 = True
-    if hostname_2.startswith('[') and hostname_2.endswith(']'):
-        hostname_2 = hostname_2[1:-1]
-        addr2_ipv6 = True
-
-    if addr1_ipv6 and addr2_ipv6:
-        try:
-            if (socket.gethostbyaddr(hostname_1)
-                    == socket.gethostbyaddr(hostname_2)):
-                return True
+    def resolve_host(name):
+        for family in (socket.AF_INET, socket.AF_INET6):
+            result = None
```

Comment:
Not needed, if 'try' executes with success, 'result' can't be unbound once try/except is finished

Reply:
It is needed

>>> try:
...   result = socket.getaddrinfo('sdfsdfsdf', None)
... except socket.gaierror:
...   pass
... 
>>> result
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'result' is not defined

Reply:
the idea is based on the fact that we just check if name resolves to localhost or some ip address, but we use this value for two string comparison, so, I think, if it is not resolving, we just return it for comparison, then it will not patch localhost and the final decision will be different hosts. Also, this way we can avoid extra exception handling and also we have less indentation

Reply:
Ok, you are right, it is nice this way.